bindings/java/src/org/sleuthkit/datamodel/blackboardutils/CommunicationArtifactsHelper.java

```diff
@@ -864,4 +869,16 @@ private void addMessageReadStatusIfKnown(MessageReadStatus readStatus, Collectio
 		}
 	}
 
+	/**
+	 * Returns self account instance.  Lazily creates it if one doesn't exist yet.
+	 * 
+	 * @return Self account instance.
+	 * @throws TskCoreException 
+	 */
+	private AccountFileInstance getSelfAccountInstance() throws TskCoreException {
```

Comment:
We should probably synchronize this method. As we drive down the call stack we eventually get to an INSERT/IGNORE, but that does not ensure visibility of the value of selfAccountInstance across all threads and making that field volatile would still leave us with a check-then-act race condition. Objects of this class were immutable and therefore thread-safe; I think we should keep it thread-safe as utility class for module developers.